sktime/transformations/base.py

```diff
@@ -1021,11 +1021,17 @@ def _convert_output(self, X, metadata, inverse=False):
             #   we cannot convert back to pd.Series, do pd.DataFrame instead then
             #   this happens only for Series, not Panel
             if X_input_scitype == "Series":
-                _, _, metadata = check_is_mtype(
+                valid, msg, metadata = check_is_mtype(
                     Xt,
                     ["pd.DataFrame", "pd.Series", "np.ndarray"],
                     return_metadata=True,
                 )
+                if not valid:
+                    raise TypeError(
+                        f"_transform output of {type(self)} does not comply "
+                        "with sktime mtype specifications. Returned error message:"
```

Comment:
possible improvement: add pointer to the mtype specifications?

Reply:
added